core/types/block.go

```diff
@@ -84,7 +86,7 @@ type Header struct {
 	GasLimit    uint64         `json:"gasLimit"         gencodec:"required"`
 	GasUsed     uint64         `json:"gasUsed"          gencodec:"required"`
 	Time        uint64         `json:"timestamp"        gencodec:"required"`
-	Extra       []byte         `json:"extraData"        gencodec:"required"`
+	Extra       []byte         `json:"extraData"        gencodec:"required" deep:"-"`
```

Comment:
What does the `deep` tag mean here?

Reply:
Can you maybe just json-encode both values, and compare the json? That way you don't have to introduce special tags or libraries.

Reply:
Agreed, I've removed this and am now checking a few individual fields in the header.